sktime/networks/cntc.py

```diff
@@ -0,0 +1,244 @@
+        rnn1 = keras.layers.Reshape((64, input_shape[1]))(rnn1)
+
+        # Combining CNN and RNN
+        conc1 = keras.layers.Concatenate(
+            axis=-2, name="contextual_convolutional_layer"
+        )([conv1, rnn1])
+
+        # Final CNN for C-CNN (WHY)
+        conv2 = keras.layers.Conv1D(
+            self.filter_sizes[1],
+            self.kernal_sizes[1],
+            activation="relu",
+            kernal_initializer="glorot_uniform",
+            name="standard_cnn_layer",
+        )(conc1)
+        conv2 = keras.layers.Dense(
+            input_shape[1],
+            input_shape=(input_shape[0], keras.backend.int_shape(conv2)[2]),
+        )(conv2)
+        conv2 = keras.layers.BatchNormalization()(conv2)
+        conv2 = keras.layers.Dropout(0.1)(conv2)
+
+        # CLSTM Arm
+        input_layers.append(keras.layers.Input(input_shape))
+        lstm1 = keras.layers.LSTM(
+            self.lstm_size * input_shape[1],
+            return_sequences=False,
+            kernal_initializer="glorot_uniform",
+            activation="relu",
+        )(input_layers[2])
+        lstm1 = keras.layers.Reshape((self.lstm_size, input_shape[1]))(lstm1)
+        lstm1 = keras.layers.Dropout(self.dropout)(lstm1)
+        merge = keras.layers.Concatenate(
+            axis=-2, name="contextual_convolutional_layer"
+        )([conv2, lstm1])
+
+        # Output calculation based on combination
+        avg = keras.layers.MaxPooling1D(pool_size=1, strides=None, padding="valid")(
+            merge
+        )
+        avg = keras.layers.Dropout(0.1)(avg)
+
+        # Adding self attention
+        att = SeqSelfAttention(
+            attention_width=10,
+            attention_activation="sigmoid",
+            name="Attention",
+            attention_type="multiplicative",
+        )(avg)
+        att = keras.layers.Dropout(0.1)(att)
+
+        # Adding ouutput MLP Layer
+        mlp1 = keras.layers.Dense(
+            self.dense_size, kernal_initializer="glorot_uniform", activation="relu"
+        )(att)
+        mlp1 = keras.layers.Dropout(0.1)(mlp1)
+        mlp2 = keras.layers.Dense(
+            self.dense_size, kernal_initializer="glorot_uniform", activation="relu"
+        )(mlp1)
+        mlp2 = keras.layers.Dropout(0.1)(mlp2)
+        flat = keras.layers.Flatten()(mlp2)
+        return input_layers, flat
+
+    def prepare_input(self, X):
+        """
+        Prepare input for the CLSTM arm of the model.
+
+        According to the paper:
+            "
+                Time series data is fed into a CLSTM and CCNN networks simultaneously
+                and is perceived differently. In the CLSTM block, the input data is
+                viewed as a multivariate time series with a single time stamp. In
+                contrast, the CCNN block receives univariate data with numerous time
+                stamps
+            "
+
+        Arguments
+        ---------
+        X: tuple,
```

Comment:
plase be more specific on the format, surely the input cannot be any tuple, and the output cannot be any tuple

Reply:
About data type specificity, I am still not very clear about them. I will do a run based on test dataset to get better idea. I have kept it empty for now for the same reason.